core/src/jvmTest/kotlin/com/ustadmobile/core/controller/PersonAccountEditPresenterTest.kt

```diff
@@ -0,0 +1,218 @@
+            import(ustadTestRule.diModule)
+            bind<UstadAccountManager>(overrides = true) with singleton { accountManager }
+            bind<UstadMobileSystemImpl>(overrides = true) with singleton { impl }
+        }
+
+        repo = di.directActiveRepoInstance()
+    }
+
+    @After
+    fun tearDown() {
+        mockWebServer.shutdown()
+    }
+
+    private fun enQueuePasswordChangeResponse(success: Boolean = true){
+        if(success){
+            mockWebServer.enqueue(MockResponse()
+                    .setResponseCode(200)
+                    .setBody(Gson().toJson(UmAccount(mPersonUid, "", "auth", "")))
+                    .setHeader("Content-Type", "application/json"))
+        }else{
+            mockWebServer.enqueue(MockResponse().setResponseCode(403))
+        }
+    }
+
+
+    private fun createPerson(withUsername: Boolean = false, isAdmin: Boolean = false): Person {
+        return Person().apply {
+            fatherName = "Doe"
+            firstNames = "Jane"
+            lastName = "Doe"
+            if(withUsername){
+                username = "dummyUserName"
+            }
+            admin = isAdmin
+            personUid = mPersonUid
+            repo.personDao.insert(this)
+        }
+    }
+
+    @Test
+    fun givenPresenterCreated_whenUsernameIsNullAndSaveClicked_shouldShowErrors(){
+        val person = createPerson(false)
+        val args = mapOf(UstadView.ARG_ENTITY_UID to person.personUid.toString())
+        val presenter = PersonAccountEditPresenter(context, args,mockView, di, mockLifecycleOwner)
+        presenter.onCreate(null)
+        presenter.handleClickSave(person)
+        argumentCaptor<Boolean>{
+            verify(mockView, timeout(defaultTimeOut).atLeastOnce()).usernameRequiredErrorVisible = capture()
+            assertEquals("Username field error was shown", true, firstValue)
+        }
+    }
+
+    @Test
+    fun givenPresenterCreated_whenUserIsNotAdminAndSaveClicked_shouldShowErrors() {
+        val person = createPerson()
+        val args = mapOf(UstadView.ARG_ENTITY_UID to person.personUid.toString())
+        val presenter = PersonAccountEditPresenter(context,args,mockView, di, mockLifecycleOwner)
+        presenter.onCreate(null)
+        presenter.handleClickSave(person)
+        argumentCaptor<Boolean>{
+            verify(mockView, timeout(defaultTimeOut).atLeastOnce()).firstPasswordFieldRequiredErrorVisible = capture()
+            assertEquals("Current password field error was shown", true, firstValue)
+        }
+    }
+
+    @Test
+    fun givenPresenterCreated_whenNewPasswordIsNotFilledAndSaveClicked_thenShouldShowError(){
+        val person = createPerson(isAdmin = true)
+        val args = mapOf(UstadView.ARG_ENTITY_UID to person.personUid.toString())
+        val presenter = PersonAccountEditPresenter(context, args,mockView, di, mockLifecycleOwner)
+        presenter.onCreate(null)
+        presenter.handleClickSave(person)
+        argumentCaptor<Boolean>{
```

Comment:
You don't need an argument captor for this kind of call. You could use just verify:

```
verify(mockView, timeout(defaultTimeout).atLeastOnce()).secondPasswordFieldRequiredErrorVisible = true
```
or use argument matchers
```
verify(mockView, timeout(defaultTimeout).atLeastOnce()).secondPasswordFieldRequiredErrorVisible = eq(true)
```